.github/workflows/lint-build-test.yml

```diff
@@ -1,14 +1,12 @@
 name: Lint, Build, and Test
 
 on:
-  push:
-    branches: [main]
-  pull_request:
+  workflow_call:
```

Comment:
This is an interesting idea. I wasn't super happy about publishing happening under a workflow called "build-lint-test", despite it functionally doing what we wanted. This resolves that naming issue.